test/Microsoft.AspNetCore.Routing.Tests/Template/TemplateMatcherTests.cs

```diff
@@ -751,12 +751,30 @@ public void TryMatch_RouteWithCatchAllClauseCapturesEmptyContent()
         }
 
         [Fact]
-        public void TryMatch_RouteWithCatchAllClauseUsesDefaultValueForEmptyContent()
+        public void TryMatch_RouteWithCatchAll_MatchesEmptyContent_DoesNotReplaceExistingRouteValue()
+        {
+            // Arrange
+            var matcher = CreateMatcher("{p1}/{*p2}");
+
+            var values = new RouteValueDictionary(new { p2 = "hello" });
+
+            // Act
+            var match = matcher.TryMatch("/v1", values);
+
+            // Assert
+            Assert.True(match);
+            Assert.Equal<int>(2, values.Count);
+            Assert.Equal("v1", values["p1"]);
+            Assert.Equal("hello", values["p2"]);
+        }
+
+        [Fact]
+        public void TryMatch_RouteWithCatchAll_UsesDefaultValueForEmptyContent()
```

Comment:
Is the name of this test accurate? You aren't checking the default value, you are checking the one that you override.